generate_variant_annotation.py

```diff
@@ -1,239 +1,114 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-#
-# Ed Mountjoy
-#
-
-import hail as hl
 import sys
+import argparse
+import logging
 
-def main():
-
-    # Args
-    version = '190129'
-    hail_table = 'gs://gnomad-public/release/2.1/ht/genomes/gnomad.genomes.r2.1.sites.ht'
-    chain_file = 'gs://hail-common/references/grch37_to_grch38.over.chain.gz'
-    cadd_table = 'gs://genetics-portal-staging/variant-annotation/extra_datasets/CADD_v1.4_GRCh37/output/cadd_v1.4_gnomad.genomes.r2.0.1.sites.ht'
-    out_parquet = 'gs://genetics-portal-staging/variant-annotation/{version}/variant-annotation.parquet'.format(version=version)
-    out_sitelist = 'gs://genetics-portal-staging/variant-annotation/{version}/variant-annotation.sitelist.tsv.gz'.format(version=version)
-    out_partitions = 256
-    maf_filter = 0.001 # 0.1%
+import hail as hl
+from hail.expr.functions import allele_type
+
+
+# Gnomad hail table:
+GNOMAD_3_TABLE = 'gs://gcp-public-data--gnomad/release/3.1.1/ht/genomes/gnomad.genomes.v3.1.1.sites.ht'
+
+# GRCh38 to 37 chainfile:
+CHAIN_FILE = 'gs://hail-common/references/grch38_to_grch37.over.chain.gz'
+
+# Parameters:
+OUT_PARTITIONS = 256
+MAF_THRESHOLD = 0.001
+
+# Population of interest:
+POPULATIONS = {
+    'afr',  # African-American
+    'amr',  # American Admixed/Latino
+    'ami',  # Amish ancestry
+    'asj',  # Ashkenazi Jewish
+    'eas',  # East Asian
+    'fin',  # Finnish
+    'nfe',  # Non-Finnish European
+    'mid',  # Middle Eastern
+    'sas',  # South Asian
+    'oth'  # Other
+}
+
+# Function to convert AF to MAF:
+def af_to_maf(af):
+    return hl.if_else(af <= 0.5, af, 1 - af)
 
-    # Local paths
-    # hail_table = 'data/gnomad.genomes.head100k.r2.1.sites.ht'
-    # chain_file = 'data/grch37_to_grch38.over.chain.gz'
-    # cadd_table = 'prepare_extra_datasets/CADD_v1.4_GRCh37/temp/cadd_v1.4_gnomad.genomes.r2.0.1.sites.ht'
-    # out_parquet = 'output/variant-annotation.parquet'
 
-    # Check output doesn't exist
-    if hl.utils.hadoop_exists(out_parquet):
-        sys.exit('Error! Output file already exists: {0}'.format(out_parquet))
+def main(gnomad_file, chain_file, maf_threshold, out_folder, test=None):
 
-    #
-    # Load ---------------------------------------------------------------------
-    #
+    # Output files:
+    out_parquet = f'{out_folder}/variant-annotation.parquet'
+    out_sitelist = f'{out_folder}/variant-annotation.sitelist.tsv.gz'
 
     # Load data
-    ht = hl.read_table(hail_table)
-    print('Total number of rows: ', ht.count())
+    ht = hl.read_table(gnomad_file)
 
-    # DEBUG take head
-    # ht = ht.head(10000)
+    # If process is being tested, take head:
+    if test:
+        ht = ht.head(test)
 
-    # Assert that all alleles are biallelic
+    # Assert that all alleles are biallelic:
     assert(ht.all(ht.alleles.length() == 2))
```

Comment:
Make the assert error informative.
```suggestion
    assert (ht.all(ht.alleles.length() == 2)), 'Monoallelic variants have been found.'
```

Reply:
I'm not merging this suggestion, as there are other cases eg. multiallelics.